### DISCUSSION TITLE
[Unknown Mod][Creatures] Adjustments for Ghosts

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#6849] [Ideas]
**Adjustments for Ghosts.**

Some changes I would like to make for the Ghosts. :)

1) When it gets the hand, a troop of ghosts (controlled by a human or an AI player) has 10% chance to automatically attack an allied troop (Ghosts in neutral cities/castles and on the adventure map are not affected).

2) If the combined power of the Ghosts (even if the Ghosts are divided in several groups) present in the army (controlled by a human or an AI player) is the greater threat (the higher damage potential in the army), or if there is only Ghosts in the army, at start of combat or during the combat (they can be a lesser threat at the start of combat, and become a greater threat during combat), then there is 10% chance the Ghosts decide to vanish (the starting animation of the Teleport spell (or a new one) can be used when this arrives), and so disappear from the combat area.
If the hero (or player) has no other troops, then he loses the combat, and the hero goes in the hero pool with the standard troops as usual.

A symbol is displayed over the sprite of the Ghosts troop (over all the Ghosts troops if several groups) in the hero screen, to inform the player that the Ghosts are the greater threat in his army, so he can make decision about it.

There will also be artifacts and spells related to Ghosts.

**Functioning of Ghosts' rules during combat.**

At the beginning of the combat, the state of the groups of Ghosts is set to "normal".
So, the heroes (or players) control the groups of Ghosts normally.

When a group of Ghosts gets the hand, there is 10% chance that the group of Ghosts go crazy.
If this happens, the state of the group of Ghosts is set to "crazy".

If the group of Ghosts, or the sum of all groups of Ghosts present in the army is the biggest threat, there is 10% chance that the group of Ghosts decides to vanish.
If this happens, the state of the group of Ghosts is set to "vanish".

If the state of the group of Ghosts is "normal", the hero (or player) controls the group of Ghosts normally, he can make them move, attack or skip.
The state of the group of Ghosts remains "normal".

If the state of the group of Ghosts is "crazy", the group of Ghosts attacks an allied troop of the army.
The state of the group of Ghosts remains "crazy".

If the state of the group of Ghosts is "vanish", the group of Ghosts decides to vanish from the battlefield.
If it was the last troop of the hero (or player), he loses the combat.
If the hero possesses the artifact "Helmet of Hopefulness", the group of Ghosts don't vanish, but the Ghosts are transformed into Peasants (ratio 1 for 1).
At the end of the combat, the group of Peasants is conserved.
If there is no free slot in the army (or if the "Game: Hero and Captain have extended inventory for creatures" option is not enabled) to receive the Peasants, the Peasants vanish.

At the beginning of each turn, if there are only groups of Ghosts present in the two heroes' armies (or players) on the battlefield, and the state of all groups of Ghosts is "crazy", then the combat ends and the two heroes (or players) lose the combat.

The Ghosts are then generated (total sum of the number of Ghosts present in all groups of Ghosts at the end of the combat) on the map at the location of the defending hero.
If the location is an active object on the map (ex: Mine, Fort, Hill Fort, Temple, Magic Well, Witch's Hut, Tree of Knowledge, etc.), the location is guarded by the Ghosts and the Haunt spell animation (or a new one) is used on the location. To enter the guarded location, the hero must fight the Ghosts.
If the combat was a siege, the town/castle becomes neutral and the Ghosts defend the town/castle (in garrison).

Ghosts' rules apply to all Ghosts directed by heroes (or players) on the battlefield (human or AI players), even those generated by artifacts and spells.
Ghosts' rules don't apply to neutral Ghosts.

**Week of the creature.**

When the "Week of the Peasant" comes, all the Ghosts controlled by a human or an AI player are transformed into Peasants.
The Ghosts in the neutral towns/castles and on the adventure map are not transformed.

This is an option:
"Adjustments for Ghosts"

**Comment by ghost** [comment]
Hmm, amazing idea of balancing the Ghosts. They are really powerfull.

But i prefer more easier adjust.

My suggestion: make souls draining rate significantly lower. From 100% to 10% or something like that.

Current state: kill 100 units = drain 100 souls.

New state: kill 100 units = gain only 10 souls.

**Reply by LeHerosInconnu** [reply]
Hello @Jerry11CIB,

> Hmm, amazing idea of balancing the Ghosts. They are really powerfull.

Thank you. :)
There is something similar in King's Bounty (1990) about the hero's leadership.
I adapted the idea for ghosts.

> But i prefer more easier adjust.
> 
> My suggestion: make souls draining rate significantly lower. From 100% to 10% or something like that.
> 
> Current state: kill 100 units = drain 100 souls.
> 
> New state: kill 100 units = gain only 10 souls.

I think this would only delay in time the absolute power of the ghosts.
When the ghosts has reached a certain number, it doesn't really matter if they convert more or less ghosts.
Also, neutral ghosts would be penalized and become less dangerous and less feared.
